Fix/update cluster changes

marcjulian opened the pull request:
Fixes #188

HarelM commented on projects/ngx-maplibre-gl/src/lib/map/map.service.ts:
```diff
@@ -44,6 +45,7 @@ import type {
   MapImageOptions,
 } from './map.types';
 import { keepAvailableObjectValues } from '../shared/utils/functions/object.fn';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
```
Is this the right path for import?

marcjulian replied:
Yes its the correct package. But I just noticed the subscription will be destroyed because it is added to the subscription list.

HarelM commented:
I've noticed that the cluster code keeps refreshing even if nothing changes (I've added a console.log in `updateCluster` method in `MarkersForClustersComponent`) do you think it's the right behavior?

marcjulian commented:
> I've noticed that the cluster code keeps refreshing even if nothing changes (I've added a console.log in `updateCluster` method in `MarkersForClustersComponent`) do you think it's the right behavior?

I added the same log in `updateCluster` before #178 was merged. The function is only triggered once at the beginning and when zoomed. I guess its not right that it keeps calling this function after #178 without any interaction.

HarelM commented:
Yup...

HarelM commented:
I wonder if there are other places where this happens.
Cc: @n-elhk.

marcjulian commented:
@HarelM I'm using now `afterRender` to call `applyChanges` instead of  `zone.onMicrotaskEmpty`. I also renamed `applyChanges` to `clearMapElements`, what do you think about the name?

I think the case where `updateCluster` is being called multiple times is unrelated with this issue, perhaps we should open a new issue and pr for that.

HarelM reviewed (approved):
Looks good to me.

HarelM commented:
Feel free to open a different issue, whatever work for you, I'm good with it.
I would simply like to make sure we solve this as this is a performance issue from my point of view.
